### PR TITLE
Python: Add support for yaml prompt template

### LINKED ISSUE #4640: Python: Provide support for YAML serialization of prompts (combine config.json and skprompt.txt)

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

SK Python doesn't have a way to create a prompt template config from a yaml file. This is neeed to further achieve parity with dotnet and java. 

### Description

This PR introduces:
- some kernel methods to load yaml prompts from a directory as well as create a kernel function from a string that contains yaml. Closes #4640 
- Unit tests and a syntax example showing how to create a yaml prompt and load it

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/README.md">Py3.8 Test Coverage</a><details><summary>Python 3.8 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py">kernel.py</a></td><td>324</td><td>29</td><td>91%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L194-L198">194&ndash;198</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L200-L204">200&ndash;204</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L206-L210">206&ndash;210</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L218-L220">218&ndash;220</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L239">239</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L245">245</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L247-L251">247&ndash;251</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L253-L257">253&ndash;257</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L259-L264">259&ndash;264</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L266">266</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L311-L315">311&ndash;315</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L332-L337">332&ndash;337</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L344">344</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L348-L352">348&ndash;352</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L365">365</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/0ee4d5d34f08e36c36845af40d402247c819f271/python/semantic_kernel/kernel.py#L626">626</a></td></tr><tr><td><b>TOTAL</b></td><td><b>5368</b></td><td><b>1004</b></td><td><b>81%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.8 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1200 | 11 :zzz: | 0 :x: | 0 :fire: | 24.106s :stopwatch: |

File: python/tests/assets/test_plugins/TestPlugin/TestFunctionYaml/test_function.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: TestFunction
+template_format: semantic-kernel
```
[the remark this answers is not in the document]
Comment (moonbox3) [reply python/tests/assets/test_plugins/TestPlugin/TestFunctionYaml/test_function.yaml:2]: Definitely, great idea, adding now

File: python/semantic_kernel/kernel.py [hunk 2]
```diff
@@ -550,27 +551,53 @@ def import_native_plugin_from_directory(self, parent_directory: str, plugin_dire
 
         return None
 
-    def import_plugin_from_prompt_directory(self, parent_directory: str, plugin_directory_name: str) -> KernelPlugin:
+    def import_plugin_from_yaml_prompt_directory(
```
[the remark this answers is not in the document]
Comment (moonbox3) [reply python/semantic_kernel/kernel.py:554]: Yes, just added this functionality.

**Review by eavanvalkenburg — APPROVED** [review]
Awesome stuff!

**Review by juliomenendez — APPROVED** [review]
LGTM

**Pull request merged**